evi-python-function-calling/README.md

```diff
@@ -0,0 +1,161 @@
+
+### Setting up a virtual environment (optional)
+
+You can create a virtual environment using either `conda` from Miniconda or Anaconda, or the built-in `venv` module in Python. Instructions for both methods are provided below.
+
+#### Using `conda`
+1. Install `conda` using [Miniconda](https://docs.anaconda.com/miniconda/), the free minimal installer for it.
+2. Create a new virtual environment:
+    ```bash
+    conda create --name evi-env python=3.11
+    ```
+3. Activate the virtual environment:
+    ```bash
+    conda activate evi-env
+    ```
+
+#### Using `venv`
+
+To create a virtual environment with `venv`, run the following commands in your terminal:
+
+1. Create a new virtual environment:
+    ```bash
+    python -m venv evi-env
+    ```
+2. Activate the virtual environment:
+    ```bash
+    source evi-env/bin/activate
+    ```
+
+### Package and system dependencies
+
+Follow these steps to install the required packages and system dependencies for using environment variables, EVI, and handling audio input/output.
+
+#### Environment Variables
+
+Install the `python-dotenv` package to load variables from a `.env` file:
+
+```bash
+pip install python-dotenv
+```
+
+#### Asynchronous HTTP requests
+
+Install the `aiohttp` package to make asynchronous HTTP requests in the weather fetching function:
+
+```bash
+pip install aiohttp
+```
+
+#### EVI and audio input/output
+
+Install the Hume Python SDK with microphone support:
+
+```bash
+pip install "hume[microphone]"
+```
+
+For audio playback and processing, additional system-level dependencies are required:
+
+##### macOS
+
+Install `ffmpeg` using Homebrew:
+
+```bash
+brew install ffmpeg
+```
+
+##### Linux
+
+Install the required packages:
+
+```bash
+sudo apt-get --yes update
+sudo apt-get --yes install libasound2-dev libportaudio2 ffmpeg
+```
+
+##### Windows
+
+Not yet supported.
+
+## EVI setup
+
+Before running this project, you'll need to set up EVI with the ability to leverage tools or call functions. Follow these steps for authentication, creating a Tool, and adding it to a configuration.
+
+1. Create a `.env` file in the root folder of the repo and add your [API Key and Secret Key](https://dev.hume.ai/docs/introduction/api-key):
+
+```sh
+HUME_API_KEY=<YOUR API KEY>
+HUME_SECRET_KEY=<YOUR SECRET KEY>
+```
+
+2. [Create a tool](https://dev.hume.ai/docs/empathic-voice-interface-evi/tool-use#setup) with the following payload:
+
+```json
+{
+  "name": "get_current_weather",
+  "description": "This tool is for getting the current weather.",
+  "parameters": "{ \"type\": \"object\", \"properties\": { \"location\": { \"type\": \"string\", \"description\": \"The city and state, e.g. San Francisco, CA\" }, \"format\": { \"type\": \"string\", \"enum\": [\"celsius\", \"fahrenheit\"], \"description\": \"The temperature unit to use. Infer this from the users location.\" } }, \"required\": [\"location\", \"format\"] }"
+}
+```
```

Comment:
Can these be ready-to-paste `curl` invocations instead of just JSON? Would save users some steps (looking up the endpoint url, remembering the name of the `X-Hume-Api-Key` header).?

Reply:
They can indeed, good callout!